docs/topics/tour/kotlin-tour-intermediate-classes-interfaces.md

```diff
@@ -269,87 +269,185 @@ For more information about interfaces and interface inheritance, see [Interfaces
 
 ## Delegation
 
-Interfaces are useful, but if your interface contains many functions, child classes may end up with a lot of 
-boilerplate code. When you only want to override a small part of your parent's behavior, you need to repeat yourself a lot.
+Interfaces are useful, but if your interface contains many functions, its child classes can end up with a lot of 
+boilerplate code. If you only want to override a small part of a class's behavior, you need to repeat yourself a lot.
 
 > Boilerplate code is a chunk of code that is reused with little or no alteration in multiple parts of a software project.
 > 
 {style="tip"}
 
-For example, let's say that you have an interface called `Drawable` that contains a number of functions and one property
+For example, let's say that you have an interface called `DrawingTool` that contains a number of functions and one property
 called `color`:
 
 ```kotlin
-interface Drawable {
-    fun draw()
-    fun resize()
-    val color: String?
+interface DrawingTool {
+    val color: String
+    fun draw(shape: String)
+    fun erase(area: String)
+    fun getToolInfo(): String
 }
 ```
 
-You create a class called `Circle` which implements the `Drawable` interface and provides implementations for all of
+You create a class called `PenTool` which implements the `DrawingTool` interface and provides implementations for all of
 its members:
 
 ```kotlin
-class Circle : Drawable {
-    override fun draw() {
-        TODO("An example implementation")
+class PenTool : DrawingTool {
+    override val color: String = "black"
+
+    override fun draw(shape: String) {
+        println("Drawing $shape using a pen in $color")
     }
-    
-    override fun resize() {
-        TODO("An example implementation")
+
+    override fun erase(area: String) {
+        println("Erasing $area with pen tool")
+    }
+
+    override fun getToolInfo(): String {
+        return "PenTool(color=$color)"
     }
-   override val color = null
 }
 ```
 
-If you wanted to create a child class of the `Circle` class which had the same behavior **except** for the value of the
-`color` property, you still need to add implementations for each member function of the `Circle` class:
+If you want to create a class similar to the `PenTool` class which has the same behavior **except** for the value of the
+`color` property, you can. First, create a new class that expects an object that implements the `DrawingTool` interface as a parameter,
```

Comment:
If you want to create a class like `PenTool` with the same behavior but a different color `value`, you can create one.

Or we'd like to mention the **except** explicitly?

Reply:
I like your version, thanks!